### PR TITLE
APPSEC-71 removes unused routes, cleanup

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 # API
 GET        /api/repository                            @uk.gov.hmrc.leakdetection.controllers.ApiController.repositories
```
Comment (hmrc-phil) [inline conf/app.routes:4]: What uses /api/repository ?

File: conf/app.routes [hunk 2]
```diff
@@ -2,7 +2,6 @@
...
-GET        /api/repository/:repository/latest         @uk.gov.hmrc.leakdetection.controllers.ApiController.latestReportsForEachBranch(repository: Repository)
 
 GET        /api/report/:id                            @uk.gov.hmrc.leakdetection.controllers.ApiController.report(id: ReportId)
```
Comment (hmrc-phil) [inline conf/app.routes:6]: I don't think anything uses this either

**Review by hmrc-phil — APPROVED** [review]
I thought I'd also noticed some methods in the leaks service/repository that were no longer needed either, but I may have been wrong!

File: conf/app.routes (hunk 1 above)
Comment (christopherjturner) [reply conf/app.routes:4]: currently catalogue uses it

**Pull request merged**